### PR TITLE
micrond: initial check-in from freifunk-gluon package feed

### PR DESCRIPTION
**Pull request by feckert**
Maintainer: @NeoRaider
Compile tested: no source change only package movement
Run tested: no source change only package movement

Description:
In OpenWrt there is no crontab for time-controlled tasks for system services like in other distributions (debian), because in OpenWrt the crontab from the busybox is used. Each timed-controlled task must edit the global user crontab file that anyone in the LuCI can see and edit (https://github.com/openwrt/packages/tree/master/net/wifischedule). Additional if we have separate files for time-controlled tasks then it is much easier and not so error prone to change the entry from the task list, because we have a dedicated file for each timed-controller task. I think the crontab should only contains entries added by the user it self and time-controlled tasks from other service should maintain there own files.
 
From the freifunk community (thanks to @NeoRaider) they have developed a small service for this -> the **micrond**. Every timed-controlled task can create a file for the **micrond** in the responsible directory.

### CODE REVIEW

**Comment by NeoRaider** [comment]
I guess I should move this to use USE_PROCD before it is merged. I'll try to get this done in the next few days.

**Comment by feckert** [comment]
@NeoRaider I changed the init script to use procd. Please have a look.

File: utils/micrond/files/etc/init.d/micrond [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+USE_PROCD=1
+
+CRONDIR=/usr/lib/micron.d
+
+service_triggers() {
+	for file in $CRONDIR/*; do
+		if [ -f /etc/config/${file##*/} ]; then
+			procd_add_reload_trigger "${file##*/}"
+		fi
+	done
```
Comment (NeoRaider) [inline utils/micrond/files/etc/init.d/micrond]: I don't think this reload trigger is useful. There is usually no relation between changes to UCI config and the need to reload crontabs. The reload_service function can be dropped as well.

File: utils/micrond/Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+define Package/micrond
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Small Cron daemon providing a cron.d directory other packages can install their crontabs into
+  DEPENDS:=
+endef
+
+define Package/micrond/install
+	$(CP) ./files/* $(1)/
```
Comment (NeoRaider) [inline utils/micrond/Makefile]: I think it would make sense to replace this $(CP) with an $(INSTALL_DIR) and touch, and get rid of the files directory.

**Comment by feckert** [comment]
@NeoRaider Should I close the pull request? Do you care that the package gets into the openwrt feed?

**Comment by rotanid** [comment]
@feckert i'd leave it open until another version of it is merged - this way it serves as a reminder

**Comment by karlp** [comment]
I commented on #4569 with a simple script that sets up daily/weekly directories for cron running, that might be enough for most uses?

**Comment by NeoRaider** [comment]
Thanks, adjusted and merged.

**Comment by feckert** [comment]
@NeoRaider thanks for merging

**Pull request closed without merging**